### PR TITLE
fix: [PL-43039]: Remove AllowOverrides header for project settings

### PR DESCRIPTION
**Pull request by tiwaritejas**
### Summary

Allow overrides section is logically incorrect for project settings as it is the lowest subscope that we support. This has been found to be misleading or confusing for some users. Hence, this PR removes the allowOverrides section for project settings

#### Screenshots

Screenshot 2023-11-22 at 5 06 19 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by tiwaritejas** [comment]
retrigger jest

**Comment by tiwaritejas** [comment]
retrigger messagecheck

File: src/modules/27-platform/default-settings/components/SettingTypeRow.tsx [hunk 1]
```diff
@@ -66,6 +66,10 @@ const getLowestAvailableScope = (allowedScopes: SettingDTO['allowedScopes'] | un
 type SettingChangedViaType = 'RESTORE' | 'UPDATE' | undefined
 
 export const SettingTypeRowHeader: React.FC = () => {
+  const { projectIdentifier, orgIdentifier, accountId } = useParams<ProjectPathProps>()
+  const currentScope = useMemo(() => {
+    return getCurrentScope({ projectIdentifier, orgIdentifier, accountId })
+  }, [projectIdentifier, orgIdentifier])
```
Comment (ishantmehtaharness) [inline src/modules/27-platform/default-settings/components/SettingTypeRow.tsx:72]: This seems a very simple check, do we need useMemo here?

**Review by ishantmehtaharness — APPROVED** [review]
we can remove useMemo
https://github.com/harness/harness-core-ui/pull/19098/files#r1401946602

But Rest looks good

**Review by sahibaHarness — APPROVED** [review]

**Review by mayankVermaHarness — APPROVED** [review]

**Pull request merged**